helix-contract/contracts/mapping-token/v2/erc20-mapping-protocol/Erc20Sub2SubBacking.sol

```diff
@@ -0,0 +1,260 @@
+        require(msg.value >= totalFee, "backing:the fee is not enough");
+        if (msg.value > totalFee) {
+            // refund fee to msgSender
+            payable(msg.sender).transfer(msg.value - totalFee);
+        }
+        return IHelixSub2SubMessageHandle(messageHandle).sendMessage{value: bridgeFee}(
+            remoteReceiveGasLimit,
+            remoteSpecVersion,
+            remoteCallWeight,
+            remoteMappingTokenFactory,
+            message);
+    }
+
+    /**
+     * @notice reigister new erc20 token to the bridge. Only owner can do this.
+     * @param token the original token address
+     * @param name the name of the original token
+     * @param symbol the symbol of the original token
+     * @param decimals the decimals of the original token
+     */
+    function register(
+        uint256 remoteReceiveGasLimit,
+        uint32  remoteSpecVersion,
+        uint64  remoteCallWeight,
+        address token,
+        string memory name,
+        string memory symbol,
+        uint8 decimals,
+        uint256 dailyLimit
+    ) external payable onlyOperator {
+        bytes memory newErc20Contract = abi.encodeWithSelector(
+            IErc20MappingTokenFactory.newErc20Contract.selector,
+            ERC20_TOKEN_TYPE,
+            token,
+            chainName,
+            name,
+            symbol,
+            decimals,
+            dailyLimit
+        );
+        uint256 messageId = _sendMessage(
+            remoteReceiveGasLimit,
+            remoteSpecVersion,
+            remoteCallWeight,
+            remoteMappingTokenFactory,
+            newErc20Contract
+        );
+        _changeDailyLimit(token, dailyLimit);
+        emit NewErc20TokenRegistered(messageId, token);
+    }
+
+    /**
+     * @notice lock original token and issuing mapping token from bridged chain
+     * @dev maybe some tokens will take some fee when transfer
+     * @param token the original token address
+     * @param recipient the recipient who will receive the issued mapping token
+     * @param amount amount of the locked token
+     */
+    function lockAndRemoteIssuing(
+        uint256 remoteReceiveGasLimit,
+        uint32  remoteSpecVersion,
+        uint64  remoteCallWeight,
+        address token,
+        address recipient,
+        uint256 amount
+    ) external payable whenNotPaused {
+        uint256 balanceBefore = IERC20(token).balanceOf(address(this));
+        require(IERC20(token).transferFrom(msg.sender, address(this), amount), "Backing:transfer tokens failed");
+        uint256 balanceAfter = IERC20(token).balanceOf(address(this));
+        require(balanceBefore.add(amount) == balanceAfter, "Backing:Transfer amount is invalid");
+        bytes memory issueMappingToken = abi.encodeWithSelector(
+            IErc20MappingTokenFactory.issueMappingToken.selector,
+            token,
+            recipient,
+            amount
+        );
+        uint256 messageId = _sendMessage(
```

Comment:
what is messageId? chain id  + lane id + message nonce?

Reply:
Now fixed by commit https://github.com/helix-bridge/contracts/pull/7/commits/51817b8b15493a56a4967499b732972e1974c81a
and use laneId + messageNonce